src/node_crypto.cc

```diff
@@ -1744,16 +1744,21 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
       }
     }
 
-    unsigned char* pub = nullptr;
-    size_t publen = EC_KEY_key2buf(ec.get(), EC_KEY_get_conv_form(ec.get()),
-                                   &pub, nullptr);
-    if (publen > 0) {
-      Local<Object> buf = Buffer::New(env, pub, publen).ToLocalChecked();
-      // Ownership of pub pointer accepted by Buffer.
-      pub = nullptr;
-      info->Set(context, env->pubkey_string(), buf).FromJust();
-    } else {
-      CHECK_NULL(pub);
+    const EC_POINT* pubkey = EC_KEY_get0_public_key(ec.get());
+    if (pubkey != nullptr) {
+      point_conversion_form_t form = EC_KEY_get_conv_form(ec.get());
+      size_t publen =
+          EC_POINT_point2oct(group, pubkey, form, nullptr, 0, nullptr);
+      if (publen != 0) {
+        MallocedBuffer<unsigned char> pub(publen);
+        publen = EC_POINT_point2oct(
+            group, pubkey, form, pub.data, pub.size, nullptr);
+        if (publen != 0) {
+          Local<Object> buf =
+              Buffer::New(env, pub.release(), publen).ToLocalChecked();
```

Comment:
Rule of Three: since there's already very similar code in two other places in this file, perhaps this can be broken out into a utility function?

Alternatively, you could stick with `EC_KEY_key2buf()` and use `Buffer::Copy()` + `OPENSSL_free()` but I'd say your current code is already an improvement.

Reply:
Oh, I hadn't noticed there was another of these. Done. (Is this using `MaybeLocal` correctly?)